ddtrace/_trace/tracer.py

```diff
@@ -278,9 +278,14 @@ def __init__(self) -> None:
             register_on_exit_signal(self._atexit)
 
         self._hooks = _hooks.Hooks()
-        atexit.register(self._atexit)
         forksafe.register_before_fork(self._sample_before_fork)
-        forksafe.register(self._child_after_fork)
+
+        # Non-global tracers require that we still register these hooks, until
+        # their usage is fully deprecated. The global one will be managed by the
+        # product protocol.
+        if not isinstance(self, Tracer):
```

Comment:
Aren't non-global tracers still of the `Tracer` type?

Reply:
That's a good point. The `_instance` attribute on `Tracer` now forces a singleton for `Tracer`. However we still allow subclasses to do whatever they want AFAIK (e.g. `DummyTracer`, but also CI Vis-related tracers etc...). So in theory we shouldn't have more than one `Tracer` instance 🤔